corehq/apps/domain/models.py

```diff
@@ -997,16 +1000,19 @@ def commtrack_settings(self):
 
     @property
     def has_custom_logo(self):
-        return (self['_attachments'] and
-                LOGO_ATTACHMENT in self['_attachments'])
+        return self.has_attachment(LOGO_ATTACHMENT)
 
     def get_custom_logo(self):
         if not self.has_custom_logo:
             return None
 
         return (
             self.fetch_attachment(LOGO_ATTACHMENT),
-            self['_attachments'][LOGO_ATTACHMENT]['content_type']
+            (
+                self['_attachments'][LOGO_ATTACHMENT]['content_type']
+                if self['_attachments']
+                else self.external_blobs[LOGO_ATTACHMENT].content_type
```

Comment:
You can use [`self.blobs`](https://github.com/dimagi/commcare-hq/blob/76b69907c78bb26b450c6fb4f3281defc3c1783a/corehq/blobs/mixin.py#L62)`[LOGO_ATTACHMENT].content_type` and remove the `if`/`else` and `_attachments` lookup.

Reply:
https://github.com/dimagi/commcare-hq/pull/21239/commits/3d9b87ad7e8996e64aeda48f2864407480841b4c